modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk_test.clj

```diff
@@ -409,30 +409,51 @@
 (defn- sync-and-assert-filtered-tables [database assert-table-fn]
   (t2.with-temp/with-temp [Database db-filtered database]
     (sync/sync-database! db-filtered {:scan :schema})
-    (doseq [table (t2/select-one Table :db_id (u/the-id db-filtered))]
-      (assert-table-fn table))))
+    (let [tables (t2/select Table :db_id (u/the-id db-filtered))]
+      (assert (not-empty tables))
+      (doseq [table tables]
+        (assert-table-fn table)))))
 
 (deftest dataset-filtering-test
   (mt/test-driver :bigquery-cloud-sdk
     (testing "Filtering BigQuery connections for datasets works as expected"
-      (testing " with an inclusion filter"
-        (sync-and-assert-filtered-tables {:name    "BigQuery Test DB with dataset inclusion filters"
-                                          :engine  :bigquery-cloud-sdk
-                                          :details (-> (mt/db)
-                                                       :details
-                                                       (assoc :dataset-filters-type "inclusion"
-                                                              :dataset-filters-patterns "a*,t*"))}
-                                         (fn [{dataset-id :schema}]
-                                           (is (not (contains? #{\a \t} (first dataset-id)))))))
-      (testing " with an exclusion filter"
-        (sync-and-assert-filtered-tables {:name    "BigQuery Test DB with dataset exclusion filters"
-                                          :engine  :bigquery-cloud-sdk
-                                          :details (-> (mt/db)
-                                                       :details
-                                                       (assoc :dataset-filters-type "exclusion"
-                                                              :dataset-filters-patterns "v*"))}
-          (fn [{dataset-id :schema}]
-            (is (not= \v (first dataset-id)))))))))
+      (mt/db) ;; force the creation of one test dataset
```

Comment:
why do we need the test dataset to be created?

Reply:
Because to run the test in isolation, the test needs at least two datasets to exist in the bigquery database. One for testing inclusion and another for exclusion. For our bigquery test database that runs in CI there are hundreds, so this is a bit pedantic. But if you were to run this test on a fresh new bigquery test database this test would fail unless you created the two datasets beforehand.

Reply:
Feels like this explanation deserves to be a comment in the code. :)